default.nix

```diff
@@ -226,7 +226,7 @@ let
       # 2020-06-26 Due to a behaviour change in neat-interpolation-0.4, we
       # require n-i >= 0.4. dontCheck helps us avoid conflicts with
       # neat-interpolation's test dependencies.
-      neat-interpolation = pkgs.haskell.lib.dontCheck super.neat-interpolation_0_5_1_1;
+      neat-interpolation = pkgs.haskell.lib.dontCheck super.neat-interpolation_0_5_1_2;
```

Comment:
It's sad that we need to bump this override all the time only to discover that CI is broken again in the next PR.

Since I got stuck in #707, I don't know a better solution though.

Reply:
We should not hang up on the Nix configuration much. Classic development works isn't it? If someone has the problem with HNix Nix configuration - it is me, since NixOS requires Nix configuration for Haskell development.

Oh, by the way, I forgot the state, indeed, do not hang-up on Nix conf, it is why I made all Nix CI builds optional already. They all are `continue-on-error: true`.

We should switch into Nixpkgs integration updates I keep talking about. We did not switched because, currently, there are none in the existence for non Nix CIs. I would make them in a couple of months, or having an evening/couple of evenings inspiration.

If it bothering you too too much, we can reduce legacy tests on Nix and increase Hackage tests.

I've tried to keep optional Nix tests as optional. They also (especially testing of older GHCs/Nix channels tests) are not relevant to release. I already pointed it several times in the past, there is a single stream from master to older Nix channels.